File: backend/src/SportsGurukul.Application/Features/UserManagement/DTOs/ContactDto.cs
namespace SportsGurukul.Application.Features.UserManagement.DTOs;

public class ContactDto
{
    public Guid Id { get; set; }
    public string? PrimaryPhoneCountryCode { get; set; }
    public string? PrimaryPhoneNumber { get; set; }
    public bool PrimaryPhoneVerified { get; set; }
    public string? SecondaryPhoneCountryCode { get; set; }
    public string? SecondaryPhoneNumber { get; set; }
    public bool SecondaryPhoneVerified { get; set; }
    public string? WebsiteUrl { get; set; }
    public string? FacebookUrl { get; set; }
    public string? TwitterUrl { get; set; }
    public string? InstagramUrl { get; set; }
    public string? LinkedInUrl { get; set; }
    public string? YouTubeUrl { get; set; }
}

File: mobile/lib/features/user/infrastructure/models/contact_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'contact_dto.freezed.dart';
part 'contact_dto.g.dart';

/// DTO matching the backend `ContactDto` schema.
@freezed
abstract class ContactDto with _$ContactDto {
  const factory ContactDto({
    required String id,
    String? primaryPhoneCountryCode,
    String? primaryPhoneNumber,
    @Default(false) bool primaryPhoneVerified,
    String? secondaryPhoneCountryCode,
    String? secondaryPhoneNumber,
    @Default(false) bool secondaryPhoneVerified,
    String? websiteUrl,
    String? facebookUrl,
    String? twitterUrl,
    String? instagramUrl,
    String? linkedInUrl,
    String? youTubeUrl,
  }) = _ContactDto;

  factory ContactDto.fromJson(Map<String, dynamic> json) =>
      _$ContactDtoFromJson(json);
}
